reana_server/status.py

```diff
@@ -377,14 +378,31 @@ def get_unschedulable_nodes(self):
         )
         return [node.metadata.name for node in nodes.items]
 
+    def get_nodes_memory(self):
+        """Get list of all node memory capacities."""
+        try:
+            nodes = current_k8s_corev1_api_client.list_node()
+            return [
+                kubernetes_memory_to_bytes(node.status.capacity["memory"])
+                for node in nodes.items
+            ]
+        except ApiException:
+            return [kubernetes_memory_to_bytes("30Gi")]
+
+    def get_total_memory(self):
+        """Get total memory from all nodes."""
+        return sum(self.get_nodes_memory())
+
     def get_memory_usage(self):
         """Get nodes memory usage."""
         result = dict()
-        nodes = current_k8s_corev1_api_client.list_node()
-        for node in nodes.items:
-            result[node.metadata.name] = {"capacity": node.status.capacity["memory"]}
-
         try:
+            nodes = current_k8s_corev1_api_client.list_node()
```

Comment:
If this function fails here, the message `"Error while calling metrics.k8s.io API."` is not correct as `list_node` is not part of the metrics API.

Reply:
You are correct, it fails with different exception than `ApiException`. However `list_node()` is the function which produces the famous `Invalid value for names, must not be None` error. I've updated the code to catch it. Also checked `kubernetes-client` issues and there is already a fix for that, but it is not released yet, so I guess we can create an issue for the future to update the k8s client with this fix

Reply:
Can you share the link?

Reply:
Here is a link to PR which was merged recently: https://github.com/kubernetes/kubernetes/pull/102159